test/higher_order_ops/test_invoke_subgraph.py

```diff
@@ -0,0 +1,120 @@
+# Owner(s): ["module: functorch"]
+# flake8: noqa: B950
+
+import torch
+import torch._dynamo
+import torch._functorch
+import torch._inductor
+import torch._inductor.decomposition
+from functorch.compile import aot_function, nop
+from torch._higher_order_ops import create_invoke_subgraph_op
+from torch.testing._internal.common_utils import run_tests, TestCase
+
+
+class TestInvokeSubgraph(TestCase):
+    def test_simple(self):
+        def gn(x, y):
+            return (torch.mul(x, y),)
+
+        def fn(x, y):
+            return create_invoke_subgraph_op(gn, (x, y))[0]
+
+        x = torch.randn(8, requires_grad=True)
+        y = torch.randn(8, requires_grad=True)
+        ref = gn(x, y)[0]
+
+        x_clone = x.clone().detach().requires_grad_(True)
+        y_clone = y.clone().detach().requires_grad_(True)
+        res = fn(x_clone, y_clone)
+
+        # Run backward
+        ref.sum().backward()
+        res.sum().backward()
+
+        self.assertEqual(ref, res)
+        self.assertEqual(x.grad, x_clone.grad)
+        self.assertEqual(y.grad, y_clone.grad)
+
+    def test_aot_function(self):
+        def gn(x, y):
+            return (torch.mul(x, y),)
+
+        def fn(x, y):
+            return create_invoke_subgraph_op(gn, (x, y))[0]
+
+        x = torch.randn(8, requires_grad=True)
+        y = torch.randn(8, requires_grad=True)
+        ref = gn(x, y)[0]
+
+        x_clone = x.clone().detach().requires_grad_(True)
+        y_clone = y.clone().detach().requires_grad_(True)
+        aot_fn = aot_function(fn, nop)
+        res = aot_fn(x_clone, y_clone)
```

Comment:
This is the only test for if we've implemented the ProxyTensor + Functionalization stuff correctly. 
- let's update gn to have some mutation on an intermediate inside it.
- the real way to convince me that those two work is some expecttest with the output graph. Are you going to do this higher up in the stack?

Reply:
Yes, I have some tests higher up in the stack. I will add expecttest.

Reply:
sgtm